GTBitsOfGood/fair-districts-ga
# Predefining Georgia counties

## Predefine GA counties + counties page

Issue Number(s): #38 

Adds new `counties` page and predefines all Georgia counties. All forms and modals that add/edit counties in some capacity and their respective API routes have been edited to accommodate this change (`newspaper`, `volunteer`, `legislator`).

Additional CSS work is needed for the `counties` page by implementing overflow scroll beyond certain `y` and potentially grouping some columns. Searching/pagination should also be considered to easily find counties in the table, but that was left out due to the scope and length of this ticket. 

### Critical Changes

**BREAKING CHANGE:** Must run `npx prisma migrate dev` to seed database with predefined counties (this might nuke existing data in your database).

### Additional information

`chakra-react-select` was used to render the dropdowns for county selects. Run `npm i` upon pulling this PR.
Data sort order after refresh (pulled from db) differs from sorting after adding/editing on all 3 pages (currently, after adding/editing, the new data replaces the old at the same index). This behavior could have been present from #15 and might be a little disorienting for the user as it can look like the data state has changed

### Related PRs

Builds off of #35.

### How to Test

- Walk through basic CRUD functionality for legislator, newspaper, and volunteer pages and see if changes to counties using dropdown are reflected on the front-end as expected.
- Verify counties page displays correct information.
